### PR TITLE
fix(core): parse_url no longer does a double parse

### LINKED ISSUE #6695: parse_urls() cut invalid wikipedia urls
Hello,

parse_urls() actually cut urls like:
http://fr.wikipedia.org/wiki/Mythe_de_l'utilisation_incompl%C3%A8te_du_cerveau

$r = preg_replace_callback('/(?<![=\/"\'])((ht|f)tps?:\/\/[^\s\r\n\t<>"\']+)/i',

doesn't allow `'` but should, i think ;)

### LINKED ISSUE #8490: parse_urls() double wraps <a> tags
Not sure how the tests are passing, but parse_urls() regexp also matches URLs wrapped in <a> tags.

I propose the following regex:

```
(?!(?:[^<]+>|[^>]+<\/a[^\w]*>))(h?[t|f]??tps*:\/\/[^\s\r\n\t<>"\'\)\(]+)
```

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #8490 and fixes #6695

### CODE REVIEW

**Review by jeabakker — APPROVED** [review]

File: engine/tests/ElggCoreRegressionBugsTest.php [hunk 1]
```diff
@@ -224,6 +224,9 @@ public function test_parse_urls() {
 			'already anchor <a href="http://twitter.com/">twitter</a> test' =>
 				'already anchor <a href="http://twitter.com/">twitter</a> test',
 
+			'already anchor with link text <a href="http://twitter.com/">http://twitter.com</a> test' =>
```
Comment (hypeJunction) [inline engine/tests/ElggCoreRegressionBugsTest.php]: We need to make sure that regex is not greedy. Needs a test with more than one anchor tag. Maybe a mix of diffetent tags. And an anchor tag within another tag.

Comment (hypeJunction) [inline engine/tests/ElggCoreRegressionBugsTest.php]: Twitter has a ljnkify library we many iseful test cases. Take a peek.

**Review by hypeJunction — CHANGES_REQUESTED** [review]

**Comment by jdalsem** [comment]
working on a new PR

**Comment by hypeJunction** [comment]
I am yet to write tests, but take a look if you need inspiration: https://github.com/hypeJunction/hypeScraper/blob/master/classes/hypeJunction/Scraper/Linkify.php

**Pull request closed without merging**